core/domain/expression_parser.py

```diff
@@ -131,11 +158,25 @@ def tokenize(expression):
         else:
             token_list.append(Token(token_text))
 
-    if sum([len(token.text) for token in token_list]) != len(expression):
-        raise Exception(
-            'Invalid syntax: Unexpected token present in the expression.')
+    # Adding '*' sign after identifiers, numbers and closing brackets if they
+    # are not followed by a valid operator.
+    final_token_list = []
+    for i in python_utils.RANGE(len(token_list)):
+        if i == len(token_list) - 1:
+            final_token_list.append(token_list[i])
```

Comment:
You can take this out of the if/else branch, right, since it's the same in both places?

In general, try to keep the "branching part" as small as possible.

Reply:
Done.